tests/util/static_check.hpp

```diff
@@ -11,20 +11,86 @@
 
 namespace util {
 
-struct static_check {
-  template <typename OperationType, typename... HandlerTypes> struct bind_left {
-    template <typename OperandType> static constexpr bool invocable(auto &&fn)
+template <typename OperandType> struct lvalue {
+  using type = std::add_lvalue_reference_t<OperandType>;
+};
+
+template <typename OperandType> struct rvalue {
+  using type = std::add_rvalue_reference_t<OperandType>;
+};
+
+template <typename OperandType> struct clvalue {
+  using type = std::add_lvalue_reference_t<std::add_const_t<OperandType>>;
+};
+
+template <typename OperandType> struct crvalue {
+  using type = std::add_rvalue_reference_t<std::add_const_t<OperandType>>;
+};
+
+template <typename OperandType> struct cvalue {
+  using type = std::add_const_t<OperandType>;
+};
+
+template <typename OperandType> struct prvalue {
+  using type = OperandType;
+};
+
+template <typename OperationType, typename OperandType> struct static_check {
```

Comment:
Please ignore all the copypasta - i can't be bothered.. this is just for tests anyways.